chore: improvements to build multiple distribution flavors

phiz71 opened the pull request:
Use profiles to manage the different dependencies in release distrib, dev distrib, EE distrib, ...

Improvement:
* Instead of using maven-dependency-plugin, rely on pom dependencies and some configuration in assembly files to build the final binaries. It allows to use `mvn versions:update-properties`, to keep distribution up to date automatically
* No need to create a tar.gz, since it will be unpack in dockerFile. Just use a folder.
<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-pdfbxwsumf.chromatic.com)
<!-- Storybook placeholder end -->
<!-- UI placeholder -->
🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/multiple-distribution/index.html)
_Notes_: The deployed app is linked to the management API of the Element Zero team's environment.
<!-- UI placeholder end -->

phiz71 committed:
chore: use dependencies to build final binaries

Instead of using maven-dependency-plugin, rely on pom dependencies and some configuration in assembly files to build the final binaries

phiz71 committed:
chore: full distributions can be only dir

No need to create a tar.gz, since it will be unpack in dockerFile.
Just use a folder.

phiz71 committed:
chore: allow to build an EE version of the distrbution

phiz71 committed:
refactor: reformat xml files

gaetanmaisse commented on gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/gravitee-apim-rest-api-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml:
```diff
@@ -30,112 +30,21 @@
         <fileSet>
-            <directory>../target/distribution/config</directory>
-            <outputDirectory>config</outputDirectory>
+            <directory>../target/gravitee-apim-rest-api-distribution-${project.version}</directory>
```
It's a bit painful to have the version in the folder name as it forces us to have 1 IntelliJ config per version. Can we remove it?

phiz71 replied:
you're right, I will change it

gaetanmaisse commented on gravitee-apim-distribution/pom.xml:
```diff
@@ -38,16 +38,15 @@
-         2. update the maven-dependency-plugin so the new plugin can be copied in the right temp folder (used in the packaging).
-            It can be apim-plugins, connectors, fetchers, policies, ... or a new kind of plugins.
-         3. update the Gateway and/or the management API assembly.xml files if necessary
+         2. add the plugin in the <dependencies> section with <type>zip</type> and <scope>runtime</scope>.
```
👌🏻 it looks clear and understandable even for non Maven expert

phiz71 replied:
🙏🏻

gaetanmaisse commented on gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/gravitee-apim-gateway-standalone-distribution-zip/src/main/assembly/plugin-assembly-standalone.xml:
```diff
@@ -29,64 +30,19 @@
         <fileSet>
-            <directory>../target/distribution/config</directory>
-            <outputDirectory>config</outputDirectory>
+            <directory>../target/gravitee-apim-gateway-distribution-${project.version}</directory>
```
Same as for the rest api:
> It's a bit painful to have the version in the folder name as it forces us to have 1 IntelliJ config per version. Can we remove it?

phiz71 replied:
you're right, I will change it

gaetanmaisse commented on gravitee-apim-rest-api/docker/Dockerfile-dev:
```diff
@@ -60,29 +58,3 @@ HEALTHCHECK --interval=5s --timeout=3s --retries=6 --start-period=30s \
-VOLUME ${GRAVITEEIO_HOME}/logs
-CMD ["./bin/gravitee"]
-HEALTHCHECK --interval=5s --timeout=3s --retries=6 --start-period=30s \
-            CMD curl --user admin:adminadmin --fail http://localhost:18083/_node/health || exit 1
```
It's a legacy thing that should have been removed earlier?

phiz71 replied:
Yes, it should be part of another PR on master. It comes from the merge of the "performance branch"

phiz71 replied:
https://github.com/gravitee-io/gravitee-api-management/pull/1073

phiz71 committed:
refactor: change full distribution final name

- keep "distribution" as the final name of the folder, to simplify IntelliJ launchers management
- rename "distribution" to the appropriate project during CircleCI build, because we need both Rest API & Gateway distribution folders.

phiz71 committed:
ci: create default license folder
